[MarcusGitty] IP

MarcusGitty opened the pull request:
# EvePro

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

EvePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~FAST~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2324s2.github.io/website/admin/ip-w4.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE!**

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
- [ ] Bug Free (coming soon)

#

If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

MarcusGitty committed:
Level-0

MarcusGitty committed:
Level-1 Echo

MarcusGitty committed:
Level-2

MarcusGitty committed:
Level-3

MarcusGitty committed:
Level-4 ToDos, Events, Deadlines

MarcusGitty committed:
A-TextUiTesting - not sure why can't work though

MarcusGitty committed:
Level-5 Error Class Made

MarcusGitty committed:
Level-6 Delete Function Added

MarcusGitty committed:
Refactor the instructions into functions for easier refactoring in the future,

Refactor the if statement into switch cases so future code editing is easier.

MarcusGitty committed:
Fixed Level-5, Eve now handle Exceptions

MarcusGitty committed:
Attempting File Storage

MarcusGitty committed:
Level 7 buggy

MarcusGitty committed:
Level-7

MarcusGitty committed:
Formatted the File System

MarcusGitty committed:
Refactored more !

MarcusGitty committed:
Level-8 Done !!! Added Local Date time Object and parser

MarcusGitty committed:
Fixed Bugged with DateTime

MarcusGitty committed:
More-OOP

MarcusGitty committed:
A-Packages

MarcusGitty committed:
Merge branch 'master' into add-gradle-support

MarcusGitty committed:
Managed to Build the Jar File From Gradle

laksherthanlife commented on src/main/java/eve/fileStorage/Storage.java:
```diff
@@ -0,0 +1,102 @@
+public class Storage {
+    private static final String directoryPath = "./data";
+    private static final String filePath = "./data/Eve.txt";
+    private static File file;
```
Do you think it would be more descriptive to name this file storageFile instead?

saiutkarsh33 replied:
Definitely - I personally do not like object names to be the same as class names. Makes it confusing for me.

MarcusGitty replied:
Good comment ! Will change it

laksherthanlife commented on src/main/java/eve/tasks/Deadline.java:
```diff
@@ -0,0 +1,40 @@
+import eve.parser.Parser;
+
+public class Deadline extends Task {
+    protected LocalDateTime by;
```
do you think it be better to have this in a noun form such as dateBy?

saiutkarsh33 replied:
Yes i agree, by might be a little hard to understand

MarcusGitty replied:
Good suggesting, thank you !

laksherthanlife commented on src/main/java/eve/tasks/Event.java:
```diff
@@ -0,0 +1,39 @@
+import eve.parser.Parser;
+
+public class Event extends Task {
+    protected LocalDateTime startString;
```
do you think it would be better to call this variable stringStart instead?

saiutkarsh33 replied:
On the contrary, I believe startString is more interepretable

MarcusGitty replied:
Hmm i might change it to startDate, because the type is no longer a string, thanks for your code review guys !!

laksherthanlife reviewed (commented):
LGTM. Just a few minor nits.

MarcusGitty committed:
A-JavaDoc

MarcusGitty committed:
Enforce Coding Standard to the Best of my Ability

MarcusGitty committed:
Enforce Coding Standard to the Best of my Ability

MarcusGitty committed:
Skeleton of Level 9 done,

Need to implement java code from Pseudo Code

saiutkarsh33 commented:
LGTM!

MarcusGitty committed:
Getting Started with Level-10

MarcusGitty committed:
Level-10 JavaFX Part 3

MarcusGitty committed:
Merge pull request #2 from MarcusGitty/Branch-A-JavaDoc

Branch a java doc

MarcusGitty committed:
Merge pull request #3 from MarcusGitty/Branch-A-CodingStandard

Enforce Coding Standard to the Best of my Ability

MarcusGitty committed:
Merge branch 'master' into Branch-Level-9

MarcusGitty committed:
Merge pull request #4 from MarcusGitty/Branch-Level-9

Skeleton of Level 9 done,

MarcusGitty committed:
Merge branch 'master' into Branch-Level-10

MarcusGitty committed:
Merge pull request #5 from MarcusGitty/Branch-Level-10

Branch level 10

MarcusGitty committed:
Implement JUnit Test That Was Missed Out

MarcusGitty committed:
Implement 2 Test Each for Deadline Class and Todo Class Using Junit

MarcusGitty commented:
> LGTM!

Thanks man, I'm trying my best !

MarcusGitty commented:
> LGTM. Just a few minor nits.

Thanks for your code review !

MarcusGitty committed:
refactor find function

MarcusGitty committed:
refactored find function

MarcusGitty committed:
Implement GUI for Eve chatbot using JavaFX

This commit improves on the graphical user interface (GUI) for the Eve chatbot, significantly enhancing the user interaction experience. Following the guidance from the SE-EDU/guides on JavaFX, this update encapsulates the journey of learning JavaFX basics and applying them to create a functional and intuitive GUI for Eve.

Key Changes:
- Implemented JavaFX setup in the project, including necessary dependencies and configuration adjustments to support GUI development.
- Developed the main application window, integrating a chat interface that allows users to interact with Duke in a more engaging and visually appealing manner.
- Created custom components for displaying user and Duke responses, ensuring a clear and coherent conversation flow within the GUI.
This enhancement follows a structured approach, starting with a thorough exploration of JavaFX through the SE-EDU/guides tutorial, before applying the acquired knowledge to design and implement the chatbot's GUI. The transition from a text-based interface to a graphical one marks a significant milestone in Eve’s evolution, offering users a more dynamic and enjoyable way to interact with the chatbot.
Currently the GUI only echo’s the User Input, in future iterations I’ll integrate it with the EVE chatbot backend.

MarcusGitty committed:
Refactored the GUI and Removed them from EVE class into it's Own GUI Folder

MarcusGitty committed:
In the Middle of Refactoring and Changing but Gotto Call it A day

Currently Stuck at Eve:getResponse

and parse:parseAndExecute

MarcusGitty committed:
Currently changing the return type of methods to Strings instead of System Out Print Lines.

MarcusGitty committed:
Managed to integrate the GUI into the whole logic, Level 10 finally done

MarcusGitty committed:
Cleaned up unused imports and unused classes

MarcusGitty committed:
Merge pull request #6 from MarcusGitty/branch-A-CodeQuality

Merging Branch Code Quality into Master

MarcusGitty committed:
Put in Assert in Eve.java, Tasklist.java and Parser.java Class

MarcusGitty committed:
Merge pull request #7 from MarcusGitty/branch-A-Assertions

Put in Assert in Eve.java, Tasklist.java and Parser.java Class

MarcusGitty committed:
Trying to implement Tag Functionality

MarcusGitty committed:
Skeloton of UG is there

MarcusGitty committed:
Add the Screenshot of the UI in readme.md

MarcusGitty committed:
Fix Bug of the Format of the Image, It points to a link whether than the Image

MarcusGitty committed:
Test to see if the links to the image in readme.md
works

MarcusGitty committed:
Update readme.md file, it is 90% complete. Just missing the section of tagging a task and formatting

MarcusGitty committed:
Updated Readme.md

MarcusGitty committed:
Merge branch 'master' into BCD-Extension

MarcusGitty committed:
Merge pull request #8 from MarcusGitty/BCD-Extension

Implement Tag Functionality to Existing Tasks

MarcusGitty committed:
Updated Readme, Left with Comestic  Changes (if Any)

MarcusGitty committed:
Removed some Coding Style Violations

MarcusGitty committed:
Change the Project Readme.md of the GIthub Landing Page

MarcusGitty committed:
Changed the Image Link of readme.md

MarcusGitty committed:
Change build.gradle

MarcusGitty committed:
Event.java: Fix bug in constructor

MarcusGitty committed:
Refactor Command.java code for coding violations

MarcusGitty committed:
Refactor Eve.java code for coding violations

MarcusGitty committed:
Refactor TaskList.java code for coding violations

MarcusGitty committed:
Refactor Main.java code for coding violations

MarcusGitty committed:
Refactor MainWindow.java code for coding violations

MarcusGitty committed:
Refactor Parser..java code for coding violations

MarcusGitty committed:
Refactor Deadline.java code for coding violations

MarcusGitty committed:
Refactor Todo.java code for coding violations

MarcusGitty committed:
Fix the bug that is caused by splitting the input with white space rather than first whitespace

MarcusGitty committed:
Change Format of outputted date time

MarcusGitty committed:
Change the toString() method of Event class

MarcusGitty committed:
Merge branch 'master' into bugfix-issue13

MarcusGitty committed:
Merge pull request #17 from MarcusGitty/bugfix-issue13

Bugfix issue13

MarcusGitty committed:
Fix Storage Issue, now Eve can store and retrive objects

MarcusGitty committed:
Refactor to pass Checkstyles

MarcusGitty committed:
Merge pull request #18 from MarcusGitty/bugfix-storage-issue-11

Bugfix storage issue 11

MarcusGitty committed:
Change project Readme.md to be accurate

MarcusGitty committed:
README.MD: Change it to look nicer

MarcusGitty committed:
ReadMe: Make it nicer

MarcusGitty committed:
README.md: Change to Reflect Final Version

MarcusGitty committed:
README.md: Change for final final version

MarcusGitty committed:
Merge pull request #19 from MarcusGitty/branch-userguide-issue-#14

Branch userguide issue #14

MarcusGitty committed:
Fix the TagCommand, realise it was broken after doing some final testing

MarcusGitty committed:
Merge pull request #20 from MarcusGitty/branch-userguide-issue-#14

Fix the TagCommand, realise it was broken after doing some final testing

MarcusGitty committed:
Refactor for Checkstyles
